phoenix-core/src/main/java/org/apache/phoenix/iterate/RoundRobinResultIterator.java

```diff
@@ -300,7 +305,11 @@ public Tuple call() throws Exception {
         private RoundRobinIterator(PeekingResultIterator itr, Tuple tuple) {
             this.delegate = itr;
             this.tuple = tuple;
-            this.numRecordsRead = 0;
+            if (tuple != null) {
```

Comment:
This is meant to be a performance fix.
The tuple was peek() -ed from the delegate, so that element is actually the first of the cached rows.
Fixing this SHOULD enable processing the 2nd and later batches parallelly, but I didn't thest this.